apps/OboeTester/app/src/main/java/com/mobileer/oboetester/ManualGlitchActivity.java

```diff
@@ -166,19 +176,45 @@ public void onTestBegan() {
     // Called on UI thread
     @Override
     protected void onGlitchDetected() {
+        if (mAutoScopeBox.isChecked()) {
+            mAutoScopeBox.setChecked(false); // stop auto drawing of waveform
+            mLastDisplayTime = 0; // force draw first glitch
+        }
         long now = System.currentTimeMillis();
+        Log.i(TAG,"onGlitchDetected: glitch");
         if ((now - mLastDisplayTime) > MIN_DISPLAY_PERIOD_MILLIS) {
             mLastDisplayTime = now;
             int numSamples = getGlitch(mWaveform);
             mWaveformView.setSampleData(mWaveform, 0, numSamples);
+            int glitchLength = getGlitchLength();
+            int[] cursors = new int[glitchLength > 0 ? 2 : 1];
+            int startOfGlitch = getSinePeriod();
+            cursors[0] = startOfGlitch;
+            if (glitchLength > 0) {
+                cursors[1] = startOfGlitch + getGlitchLength();
+            }
+            mWaveformView.setCursorData(cursors);
+            Log.i(TAG,"onGlitchDetected: glitch, numSamples = " + numSamples);
             mWaveformView.postInvalidate();
         }
     }
-
-    private float[] getGlitchWaveform() {
-        return mWaveform;
+    @Override
+    protected void maybeDisplayWaveform() {
```

Comment:
Should we hide the old waveform in some cases or should we leave it as is? If we leave it as is, is there any indicator that the waveform isn't the most recent?

Reply:
I want to leave the glitch waveform on the screen so we can study it and maybe take a screenshot.

If you enable the "Auto draw" then it will update about twice per second.

> is there any indicator that the waveform isn't the most recent?

Only that the display is not changing for a long time.